stan/math/prim/fun/svd_U.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef STAN_MATH_PRIM_FUN_SVD_U_HPP
+#define STAN_MATH_PRIM_FUN_SVD_U_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Given input matrix m, return matrix U where \mathbf{m} = \mathbf{UDV^{T}}
+ *
+ * @tparam EigMat type of the matrix
+ * @param m MxN input matrix
+ * @return Orthogonal matrix U
+ */
+template <typename EigMat,
+	  require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+	  require_not_vt_var<EigMat>* = nullptr>
+Eigen::Matrix<value_type_t<EigMat>, Eigen::Dynamic, Eigen::Dynamic>
+svd_U(const EigMat& m) {
```

Comment:
This prim implementation handles all the non-reverse mode autodiff stuff and the non-autodiff stuff.

So in terms of scalar types, that is `double`, `fvar<double>`, `fvar<var>`, `fvar<fvar<double>>` and `fvar<fvar<var>>`